pkg/spdx/implementation.go

```diff
@@ -86,8 +86,22 @@ func (di *spdxDefaultImplementation) ExtractTarballTmp(tarPath string) (tmpDir s
 	}
 	defer f.Close()
 
+	// Read the first bytes to determine if the file is compressed
+	var sample [2]byte
```

Comment:
same observations as above ... do we want to put this in a method that can be called from both places?

Reply:
Ah, yes I'm aware of this. I'm working on a PR to move this functionality to [our tar package](https://github.com/kubernetes-sigs/release-utils/blob/main/tar/tar.go) and once we cut a release there I will refactor all the tar code in the project.